compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1322,6 +1324,11 @@ object Parsers {
         atSpan(in.skipToken()) { Return(if (isExprIntro) expr() else EmptyTree, EmptyTree) }
       case FOR =>
         forExpr()
+      case LBRACKET =>
+        val start = in.offset
+        val tparams = typeParamClause(ParamOwner.TypeParam)
+        assert(isIdent && in.name.toString == "->", "Expected `->`")
```

Comment:
This should be checked syntax, not an assert.

Reply:
Done.